File: Golang/190_reverseBits.go
package main

import (
	"fmt"
)


func reverseBits(n int) int {
    var res int = 0
	for i := 0; i < 32; i++ {
		res = (res << 1) | (n & 1)
		n >>= 1
	}
	return  res
}



func main() {
	// Case 1. 
	// Input: n = 43261596
	// Output: 964176192
	n := 43261596
	fmt.Print("Case 1: ", reverseBits(n))
}

File: Python/190_reverseBits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        for _ in range(32):
            # 取出 n 的最低為並加到 res 的最高位裡
            res = (res << 1) | (n & 1)
            # 將 n 向右移一位
            n >>= 1
        return res
    

# Example 1:
# Input: n = 43261596
# Output: 964176192
res = Solution().reverseBits(n=43261596)
print(res)
